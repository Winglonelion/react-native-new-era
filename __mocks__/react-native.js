import { Animated as RNAnimated } from 'react-native';

export const Animated = {
  ...RNAnimated,
  parallel: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
  timing: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
};
