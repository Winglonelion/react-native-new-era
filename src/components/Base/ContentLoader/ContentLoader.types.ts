import { SvgProps } from 'react-native-svg';

export interface IContentLoaderProps extends SvgProps {
  animate?: boolean;
  backgroundColor?: string;
  foregroundColor?: string;
  rtl?: boolean;
  speed?: number;
  interval?: number;
  uniqueKey?: string;
}
