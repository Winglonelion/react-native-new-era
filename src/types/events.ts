import { GestureResponderEvent } from 'react-native';

export type PressEvent = (event: GestureResponderEvent) => any;
