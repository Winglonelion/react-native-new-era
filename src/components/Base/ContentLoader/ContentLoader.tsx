import * as React from 'react';
import { Circle, Path, Rect } from 'react-native-svg';

import { IContentLoaderProps } from './ContentLoader.types';
import ReactContentLoaderFacebook from './presets/FacebookStyle';
import NativeSvg from './Svg';

const ContentLoader: React.FC<IContentLoaderProps> = props =>
  props.children ? (
    <NativeSvg {...props} />
  ) : (
    <ReactContentLoaderFacebook {...props} />
  );

export { Circle, Rect, Path };

export default ContentLoader;
