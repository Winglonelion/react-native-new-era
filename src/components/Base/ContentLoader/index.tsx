import ContentLoader from './ContentLoader';

export { Circle, Rect, Path } from './ContentLoader';
export default ContentLoader;
