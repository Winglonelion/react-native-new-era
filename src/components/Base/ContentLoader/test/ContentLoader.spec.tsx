import React from 'react';
import { shallow } from 'enzyme';
import ContentLoader from '../ContentLoader';
import { IContentLoaderProps } from '../ContentLoader.types';
import { View } from 'react-native';

const setup = (
  propOverrides: IContentLoaderProps = {} as IContentLoaderProps,
) => {
  const props = { ...propOverrides };
  const wrapper = shallow(
    <ContentLoader {...props}>
      <View />
    </ContentLoader>,
  );
  return {
    props,
    wrapper,
  };
};

describe('ContentLoader check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ContentLoader check snapshot with props', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({
      animate: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
