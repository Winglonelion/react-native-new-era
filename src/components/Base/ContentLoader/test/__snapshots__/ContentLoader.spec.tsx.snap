// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoader check snapshot should match snapshot 1`] = `
<NativeSvg
  animate={true}
  backgroundColor="#f5f6f7"
  foregroundColor="#eee"
  interval={0.25}
  rtl={false}
  speed={1.2}
  style={Object {}}
>
  <View />
</NativeSvg>
`;

exports[`ContentLoader check snapshot with props should match snapshot 1`] = `
<NativeSvg
  animate={true}
  backgroundColor="#f5f6f7"
  foregroundColor="#eee"
  interval={0.25}
  rtl={false}
  speed={1.2}
  style={Object {}}
>
  <View />
</NativeSvg>
`;
